src/gravel/controllers/nodes/disks.py

```diff
@@ -0,0 +1,135 @@
+# project aquarium's backend
+# Copyright (C) 2021 SUSE, LLC.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+from enum import Enum
+from logging import Logger
+from typing import List, Optional
+from fastapi.logger import logger as fastapi_logger
+from pydantic import BaseModel, Field
+
+from gravel.cephadm.models import NodeInfoModel, VolumeDeviceModel
+from gravel.controllers.gstate import GlobalState
+
+
+logger: Logger = fastapi_logger
+
+
+class DiskTypeEnum(int, Enum):
+    NONE = 0
+    HDD = 1
+    SSD = 2
+
+
+class DiskInfoModel(BaseModel):
+    vendor: str = Field("", title="disk's manufacturer")
+    model: str = Field("", title="disk's model")
+
+
+class DiskModel(BaseModel):
+    path: Optional[str] = Field(None, title="disk's device path")
+    size: int = Field(0, title="size in bytes")
+    type: DiskTypeEnum = Field(DiskTypeEnum.NONE, title="disk type")
+    info: DiskInfoModel = Field(title="additional info about disk")
+
+
+class RejectedDiskModel(BaseModel):
+    disk: DiskModel = Field(title="rejected disk")
+    reasons: List[str] = Field([], title="reasons for rejection")
+
+
+class DiskSolution(BaseModel):
+    systemdisk: Optional[DiskModel] = Field(None, title="chosen system disk")
+    storage: List[DiskModel] = Field([], title="chosen storage disks")
+    storage_size: int = Field(0, title="total size for storage, in bytes")
+    rejected: List[RejectedDiskModel] = Field([], title="rejected disks")
+    possible: bool = Field(False, title="deployment possible")
+
+
+def _device_to_disk(device: VolumeDeviceModel) -> DiskModel:
+
+    def _get_disk_type(rotational: bool) -> DiskTypeEnum:
+        return DiskTypeEnum.HDD if rotational else DiskTypeEnum.SSD
+
+    return DiskModel(
+        path=device.path,
+        size=device.sys_api.size,
+        type=_get_disk_type(device.sys_api.rotational),
+        info=DiskInfoModel(
+            vendor=device.sys_api.vendor,
+            model=device.sys_api.model
+        )
+    )
+
+
+class Disks:
```

Comment:
(nit) Why does this need to be a class? Since we're not creating any instances wouldn't it make more sense to just have gen_solution as a standard method?

Reply:
Mostly future looking, that's all.